test/e2e/run-all.sh

```diff
@@ -5,72 +5,88 @@ set -e
 set -u
 set -o pipefail
 
+retry () {
+  retry=0
+  limit="${METAMASK_E2E_RETRY_LIMIT:-5}"
+  while [[ $retry -lt $limit ]]
+  do
+    "$@" && break
+    retry=$(( retry + 1 ))
+    sleep 1
+  done
+
+  if [[ $retry == "$limit" ]]
+  then
+    exit 1
+  fi
+}
+
 export PATH="$PATH:./node_modules/.bin"
 
-mocha --no-timeouts test/e2e/tests/*.spec.js
+retry mocha --no-timeouts test/e2e/tests/*.spec.js
```

Comment:
Ideally we'd retry each of these individually, rather than as a group 🤔

Reply:
Updated